spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/procedures/RewriteDataFilesProcedure.java

```diff
@@ -184,6 +184,9 @@ private RewriteDataFiles checkAndApplyStrategy(
                 .toArray(String[]::new);
         return action.zOrder(columnNames);
       } else {
+        if (sortOrderFields.isEmpty()) {
+          return action.sort();
```

Comment:
this will set the sort order to null (unintialized), which in turn uses the table sort order here. 

https://github.com/apache/iceberg/blob/master/core/src/main/java/org/apache/iceberg/actions/SortStrategy.java#L76

Reply:
we cam avoid another level of indentation if we use an else if

```java
  else if (sortOrdserFields.nonEmpty) {
       return action.sort(buildSortOrder(sortOrderFields, schema));
  } else {
       return action.sort()
  }
```

Reply:
fixed.